ansible/roles/tower_demo/tasks/main.yml

```diff
@@ -0,0 +1,113 @@
+---
+- name: install pip via yum
+  yum:
+   name: python-pip
+
+- name: Install Tower cli
+  pip:
+   name: 'ansible-tower-cli'
+   version: '3.1.8'
+
+- name: Configure the tower cli file
+  template:
+    src: "{{ ANSIBLE_REPO_PATH }}/files/tower_cli.j2"
+    dest: "~/.tower_cli.cfg"
+    mode: 0640
+
+- name: copy the license file
+  copy:
+   src: "{{ ANSIBLE_REPO_PATH }}/files/tower_license"
+   dest: "/tmp/tower_license"
+
+- name: copy the private key
+  copy:
+    src: "{{ ANSIBLE_REPO_PATH }}/workdir/{{ guid + 'key' }}"
+    dest: /tmp/id_rsa
+    mode: 0640
+  
+- name: generate ansible hosts file
+  template:
+    src: "{{ ANSIBLE_REPO_PATH }}/configs/{{ env_type }}/files/hosts_template.j2"
+    dest: "/etc/ansible/hosts"
+
+- name: Add the license of tower
+  command: |
+     tower-cli setting modify LICENSE '{"eula_accepted" : "true", "company_name": "Red Hat Internal", "contact_email": "ber@redhat.com", "contact_name": "Jon Bersuder",  "instance_count": 16, "license_date": 1514782800, "license_key": "2e2995c0e36fa1846c34db031138d44e15dcf5b04c414c462d159f2648f579aa", "license_type": "enterprise", "subscription_name": "Ansible Tower by Red Hat, Standard (16 Managed Nodes)",  "trial": false }' -u admin -p "{{ tower_admin_password }}"
+
+
+- name: Add tower scm credential
+  shell: |
+        tower-cli credential create --organization="{{ org_name }}" --kind=scm --username=test --password=test --name="{{ scm_cred_name }}"
+
+- name: Create Tower project
+  shell: |
+        tower-cli project create --organization="{{ org_name }}" --scm-type=git --scm-branch="{{ scm_branch }}" --scm-credential="{{ scm_cred_name }}" --scm-url="{{ scm_url }}" --scm-update-on-launch=True -n "{{ proj_name }}"
+
+- name: create the inventory
+  shell: |
+       tower-cli inventory create  -n "{{ inv_name }}" --organization="{{ org_name }}"
+
+- name: Add the servers to inventory
+  shell: tower-manage inventory_import --inventory-name={{ inv_name }} --source=/etc/ansible/hosts
+
+- name: create the user ssh credential
+  shell: |
+       tower-cli credential create --name="{{ ssh_cred_name }}" --username="{{ user_cred_name }}" --organization="{{ org_name }}" --kind=ssh --ssh-key-data="{{ user_cred_path }}" --become-method=sudo
+
+- name: Create the job template
+  shell:  |
+        tower-cli job_template create --name 3tier_app_demo  --project="{{ proj_name }}" --inventory="{{ inv_name }}" --job-type=run --playbook='{{ playbook_name }}' --machine-credential="{{ ssh_cred_name }}"
+
+#- name: Add tower scm credentail
```

Comment:
Are those commented lines the ansible version of the previous shell tasks ?

If yes, i think we should prefer the ansible-like task